packages/gitbook/src/components/Search/SearchInput.tsx

```diff
@@ -0,0 +1,148 @@
+    isOpen: boolean;
+    className?: string;
+}
+
+/**
+ * Input to trigger search.
+ */
+export const SearchInput = React.forwardRef<HTMLDivElement, SearchButtonProps>(
+    function SearchInput(props, ref) {
+        const { onChange, onKeyDown, onFocus, value, withAsk = false, isOpen, className } = props;
+        const inputRef = useRef<HTMLInputElement>(null);
+
+        const language = useLanguage();
+        const buttonStyles = useClassnames(['ButtonStyles']);
+
+        // Size classes for medium size button
+        const sizeClasses = ['text-sm', 'px-3.5', 'py-1.5', 'circular-corners:px-4'];
+
+        useEffect(() => {
+            if (isOpen) {
+                inputRef.current?.focus();
+                if (value) {
+                    // Place cursor at the end of the input
+                    inputRef.current?.setSelectionRange(value.length, value.length);
+                }
+            } else {
+                inputRef.current?.blur();
+            }
+        }, [isOpen]);
+
+        return (
+            <div className="relative flex size-9 grow">
+                <div
+                    ref={ref}
+                    onClick={onFocus}
+                    onKeyDown={(e) => {
+                        if (e.key === 'Enter' || e.key === ' ') {
+                            onFocus();
+                        }
+                    }}
+                    className={tcls(
+                        // Apply button styles
+                        buttonStyles,
+                        variantClasses.header,
+                        sizeClasses,
+                        // Additional custom styles
+                        'has-[input:focus]:-translate-y-px h-9 grow cursor-pointer px-2.5 has-[input:focus]:bg-tint-base depth-subtle:has-[input:focus]:shadow-lg depth-subtle:has-[input:focus]:shadow-primary-subtle has-[input:focus-visible]:ring-2 has-[input:focus-visible]:ring-primary-hover md:cursor-text',
+                        'theme-bold:has-[input:focus-visible]:bg-header-link/3 theme-bold:has-[input:focus-visible]:ring-header-link/6',
+                        'theme-bold:before:absolute theme-bold:before:inset-0 theme-bold:before:bg-header-background/7 theme-bold:before:backdrop-blur-xl ', // Special overlay to make the transparent colors of theme-bold visible.
+                        'relative z-30 max-md:absolute max-md:right-0',
+                        className
+                    )}
+                >
+                    {value ? (
+                        <Button
+                            variant="blank"
+                            label="Clear"
+                            size="medium"
+                            iconOnly
+                            icon="circle-xmark"
+                            className="-mx-1.5 animate-scaleIn px-1.5 theme-bold:text-header-link theme-bold:hover:bg-header-link/3"
+                            onClick={() => {
+                                onChange({
```

Comment:
This looks bad

Either you change the value of the input with `inputRef.current?.value = '';`

or you change the `onChange` to be `onChange: (value: string) => void`